crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_numeric_union.rs

```diff
@@ -37,6 +44,13 @@ use crate::checkers::ast::Checker;
 /// def foo(x: float | str) -> None: ...
 /// ```
 ///
+/// ## Fix safety
+/// This rule's fix is marked as safe for most cases; however, the fix will
+/// nested unions type expressions into a single top-level union.
+///
+/// The fix is marked as unsafe when comments are present within the type
+/// expression.
```

Comment:
What's the reason that the fix is marked as unsafe when comments are present? Does it produce invalid syntax? Are there cases it can't handle?

I think our general practice has now been to not mark a fix as unsafe just because it removes comments. Instead, we stated that the fix could remove comments.